sunpy/net/dataretriever/sources/ace.py

```diff
@@ -0,0 +1,336 @@
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+
+
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+from sunpy.time import TimeRange
+
+__all__ = ['SWEPAMClient', 'EPAMClient', 'MAGClient', 'SISClient']
+
+class SWEPAMClient(GenericClient):
+    """
+    Returns a list of URLS to ACE SWEPAM files corresponding to value of input timerange.
+    URL source: `ftp://ftp.swpc.noaa.gov/pub/lists/ace/`.
+    The earliest date available is from 29-Jul-2015
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'swepam'
+    
+    Returns
+    -------
+    urls: list
+          list of urls corresponding to requested time range.
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(a.Time('2015/12/28 00:00:00', '2015/12/28 00:03:00'), a.Instrument('swepam'))
+    >>> print(results)
+    >>> [<Table length=3>
+         Start Time           End Time      Source Instrument
+            str19               str19         str3     str6   
+    ------------------- ------------------- ------ ----------
+    2016-05-18 00:00:00 2016-05-19 00:00:00    ACE     swepam
+    2016-05-19 00:00:00 2016-05-20 00:00:00    ACE     swepam
+    2016-05-20 00:00:00 2016-05-21 00:00:00    ACE     swepam]
+    
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """ returns list of urls corresponding
+        to given TimeRange. """
+
+        START_DATE = datetime.datetime(2015, 7, 29)
+        if timerange.start < START_DATE:
+            raise ValueError('Earliest date for which SWEPAM data is available is {:%Y-%m-%d}'.format(START_DATE))
+        base_url = 'ftp://ftp.swpc.noaa.gov/pub/lists/ace/'
+        total_days = (timerange.end - timerange.start).days + 1
+        all_days = timerange.split(total_days)
+        result = list()
+        for day in all_days:
+            url = base_url + '{date:%Y%m%d}_ace_swepam_1m.txt'.format(date = day.end)
+            result.append(url)
+        if ((datetime.datetime.now() - timerange.end).days == 0):
+            url = base_url + 'ace_swepam_1m.txt'
+            result.append(url)
```

Comment:
Yes, If you just query a single date (which is not the current day, as in right now) it will download that day's data (which is expected). But if you query the current date (today's date, right now), it will download the real time data (updated every 5 minutes) as well as data for the current day (which is kept like any other file.)